docs/snapshotting/random-for-clones.md

```diff
@@ -114,22 +114,23 @@ the read result via bind mounting another file on top of
   important, bind mount another file on top of it.
 * If microVMs run on machines with IvyBridge or newer Intel processors
   (which provide RDRAND; in addition, RDSEED is offered starting with
-  Broadwell), and the noise implicitly added when random values are
-  generated is considered sufficient, then nothing else has to be done.
-* Otherwise, the conservative approach is to do the following (before
-  customer code continues its run in the clone):
+  Broadwell). Hardware supported reseeding is done on a cadence defined
+  by the Linux Kernel and should be sufficient for most cases.
+* To be as safe as possible, the direct approach is to do the following (before
+  customer code is resumed in the clone):
   1. Open one of the special devices files (either `/dev/random` or
      `/dev/urandom`).
-  1. Issue an `RNDCLEARPOOL ioctl` (requires `CAP_SYS_ADMIN`). This
-     clears and sets the entropy pools to the initial state. Should also
-     cause the reinitialization of the `/dev/urandom` `CSPRNG`.
-  1. Issue an `RNDADDENTROPY ioctl` (requires `CAP_SYS_ADMIN`) to mix
-     the provided bytes into the input entropy pool and increase the
-     entropy count. This should also cause the `/dev/urandom` `CSPRNG`
-     to be reseeded. The bytes can be generated locally in the guest,
-     or obtained from the host. Starting with version 4.17 of the
-     kernel, there’s a new ioctl request (`RNDRESEEDCRNG`) that
+  2. Issue a `RNDRESEEDCRNG` system call.
+     `RNDRESEEDCRNG` ([4.14][5], [5.10][6], (requires `CAP_SYS_ADMIN`)) that
      specifically causes the `CSPRNG` to be reseeded from the input pool.
+     1. An alternative would be to issue an `RNDADDENTROPY ioctl`
```

Comment:
See above, not sure if this should be an "alternative" or mandatory step (as per initial recommendations). Do you have any more context here?

Reply:
Yea I think you are spot on. I misunderstood the [crng_reseed()](https://elixir.bootlin.com/linux/v4.14.295/source/drivers/char/random.c#L669) call as accommodating entropy increase as well. the random.c file has been heavily refactored. I will revert on this specific point to make use of RNDADDENTROPY. The RNDADDENTROPY ioctl call doesn't seem to necessarily reseed though, would you agree that forcing a reseed would be prudent afterwards as well?